### PR TITLE
[file-sd-part-2] Query e2e test uses different service discovery options

### PR DESCRIPTION
**Pull request by ivan-valkov**
ref: https://github.com/improbable-eng/thanos/issues/492
## Changes

The query e2e test now runs 3 times with different service discovery options:
* only gossip
* only static flag configuration
* only file sd

## Verification
The tests pass

### CODE REVIEW

File: test/e2e/query_test.go [hunk 1]
```diff
@@ -17,9 +17,38 @@ import (
...
+			useFileSD:           true,
+		},
+		"all": {
+			useGossip:           true,
+			useStaticStoresFlag: true,
+			useFileSD:           true,
+		},
+	} {
+		//TODO(ivan): delete this
+		fmt.Printf("Starting TestQuerySimple with service discovery: %v\n", name)
```
Comment (bwplotka) [inline test/e2e/query_test.go]: if ok := t.Run(t, name, "" ) {

}

File: test/e2e/query_test.go [hunk 2]
```diff
@@ -17,9 +17,38 @@ import (
 	"github.com/prometheus/common/model"
 )
 
+func TestQuery(t *testing.T) {
+	for name, conf := range map[string]sdConfig{
```
Comment (bwplotka) [inline test/e2e/query_test.go]: pkg/compact/retention_test.go:21

File: test/e2e/query_test.go [hunk 3]
```diff
@@ -15,56 +15,63 @@ import (
...
+			fileSDSuite,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			testQuerySimple(t, tt)
+		})
+	}
+}
+
 // TestQuerySimple runs a setup of Prometheus servers, sidecars, and query nodes and verifies that
```
Comment (domgreen) [inline test/e2e/query_test.go]: nit update comment ot new name

File: test/e2e/query_test.go [hunk 4]
```diff
@@ -197,3 +204,18 @@ func queryPrometheus(ctx context.Context, ustr string, ts time.Time, q string, d
 	}
 	return m.Data.Result, nil
 }
+
+func defaultPromConfig(name string, replicas int) string {
```
Comment (domgreen) [inline test/e2e/query_test.go]: What is this? Im abit confused as to why we have this in the code? Is it copypasta?

**Pull request merged**